src/app/modules/item/components/item-permissions/item-permissions.component.ts

```diff
@@ -1,13 +1,62 @@
-import { Component, Input } from '@angular/core';
+import { Component, EventEmitter, Input, OnChanges, Output } from '@angular/core';
 import { ItemData } from '../../services/item-datasource.service';
 import { WatchedGroup } from 'src/app/core/services/group-watching.service';
+import {
+  ProgressSelectValue
+} from '../../../shared-components/components/collapsible-section/progress-select/progress-select.component';
+import { generateCanViewValues } from '../permissions-edit-dialog/permissions-edit-dialog-texts';
+import { Permissions } from '../../../../shared/helpers/group-permissions';
+import { GroupPermissionsService } from '../../../../shared/http-services/group-permissions.service';
+import { ActionFeedbackService } from '../../../../shared/services/action-feedback.service';
+import { PermissionsInfo } from '../../helpers/item-permissions';
 
 @Component({
   selector: 'alg-item-permissions',
   templateUrl: './item-permissions.component.html',
   styleUrls: [ './item-permissions.component.scss' ],
 })
-export class ItemPermissionsComponent {
+export class ItemPermissionsComponent implements OnChanges {
+  @Output() changed = new EventEmitter<void>();
+
   @Input() itemData?: ItemData;
   @Input() watchedGroup?: WatchedGroup;
+
+  canViewValues: ProgressSelectValue<string>[] = generateCanViewValues('Groups');
+  permissionsDialog: 'closed' | 'opened' = 'closed';
+  watchedGroupPermissions?: PermissionsInfo & {canMakeSessionOfficial: boolean};
+
+  constructor(private groupPermissionsService: GroupPermissionsService, private actionFeedbackService: ActionFeedbackService) {
+  }
+
+  ngOnChanges(): void {
+    this.watchedGroupPermissions = this.itemData?.item?.watchedGroup?.permissions ? {
+      ...this.itemData.item.watchedGroup.permissions,
+      canMakeSessionOfficial: false,
```

Comment:
Initializing the form with an arbitrary `canMakeSessionOfficial` is problematic, for any `true` initial value if a manager edits permissions through this dialog then the `canMakeSessionOfficial` will be set to `false`, resulting in an unintentional change...
According to the doc, this property is absent ([doc](https://france-ioi.github.io/algorea-devdoc/api/#operation/groupTeamProgress)), it might be an omission, can you check if it's present in the response? If not we'll have to find another solution

Reply:
The same description as above

Reply:
> The alg-permissions-edit-dialog component expects Group Permission type but watchedGroup has Item Permission type which doesn’t content canMakeSessionOfficial
By conversation with Damien:
“canMakeSessionOfficial ” is something which is not really used for now. Can ignore it